### PR TITLE
feat(user email): admin can configure to display user email

### PR DESCRIPTION
**Pull request by aeng**
https://zanata.atlassian.net/browse/ZNTA-879

API
- project contributors list - when configuration for display user set to true
- user service (private api for editor, /user/{username}) -  when configuration for display user set to true

UI
- user profile page - when logged in and configuration for display user set to true

### CODE REVIEW

**Comment by djansen-redhat** [comment]
Can I suggest making it look more like the option "Email log" below it? e.g. large text and help text visible

**Comment by aeng** [comment]
@djansen-redhat done.

**Comment by djansen-redhat** [comment]
The failure in build 5279 can be ignored.

File: zanata-war/src/main/java/org/zanata/rest/editor/service/resource/UserResource.java [hunk 1]
```diff
@@ -59,7 +59,7 @@
      *
      * @param username
      */
-    public User generateUser(String username);
+    public User generateUser(String username, boolean includeEmail);
```
Comment (huangp) [inline zanata-war/src/main/java/org/zanata/rest/editor/service/resource/UserResource.java]: I know this is already there but this method and the other one is not REST methods. They can stay private in the implementation class

File: zanata-war/src/main/java/org/zanata/rest/editor/service/UserService.java [hunk 2]
```diff
@@ -69,7 +74,7 @@ public Response getUserInfo(String username) {
     }
 
     @Override
-    public User generateUser(String username) {
+    public User generateUser(String username, boolean includeEmail) {
```
Comment (huangp) [inline zanata-war/src/main/java/org/zanata/rest/editor/service/UserService.java]: Why not just leave the method unchanged and use applicationConfiguration.isDisplayUserEmail() to get the boolean value again.

**Comment by huangp** [comment]
:+1:

**Comment by djansen-redhat** [comment]
"transferToUser(org.zanata.model.HAccount,boolean) has private access in org.zanata.rest.editor.service.UserService"

**Comment by djansen-redhat** [comment]
These seems to work fine - can I just confirm that the maintainer list placement of email was removed from the spec?

**Comment by djansen-redhat** [comment]
:white_check_mark: Tested

**Pull request merged**